lib/amqp/application/channel.ex

```diff
@@ -83,9 +83,12 @@ defmodule AMQP.Application.Channel do
   """
   @spec get_channel(binary | atom) :: {:ok, Channel.t()} | {:error, any}
   def get_channel(name \\ :default) do
-    case GenServer.call(get_server_name(name), :get_channel) do
-      nil -> {:error, :channel_not_ready}
-      channel -> {:ok, channel}
+    with false <- whereis(name) |> is_nil(),
```

Comment:
1. does whereis(name) work? shouldn't we use `get_server_name(name)`?
2. can you consider using the same structure I suggested for get_connection?

Reply:
1. As mentioned in a previous comment:
whereis(name) is a local function in the module that does the following:
```elixir
  def whereis(name) do
    name
    |> get_server_name()
    |> GenServer.whereis()
  end
```
2. done